.github/workflows/infrastructure.yml

```diff
@@ -1,76 +1,267 @@
-    - '.github/workflows/infrastructure.yml'
+    branches:
+      - infrastructure
+
+    paths: [.github/workflows/infrastructure.yml]
 
 env:
-  product: exip
+  PRODUCT: exip
+  ENVIRONMENT: infrastructure
+  TIMEZONE: 'Europe/London'
+  # Deployment environment target i.e., `development`, `staging`, `production`
+  TARGET: ${{ vars.ENVIRONMENT }}
 
 jobs:
+  # 1. Setup infrastructure variables
+  setup:
+    name: Setup 🔧
+    runs-on: [self-hosted, EXIP, infrastructure]
+    outputs:
+      environment: ${{ env.ENVIRONMENT }}
+      timezone: ${{ env.TIMEZONE }}
+    steps:
+      - name: Environment 🧪
+        run: echo "Environment set to ${{ env.ENVIRONMENT }}"
+
+      - name: Timezone 🌐
+        run: echo "Timezone set to ${{ env.TIMEZONE }}"
 
-# Development environment
-  development:
-    name: Development 🚀
-    environment: dev
-    runs-on: [ self-hosted, linux, deployment, infrastructure ]
-    env:
-      type: Pre-production
-      plan: dev
-      acr: ${{ secrets.ACR_SERVER }}
-      acr_username: ${{ secrets.ACR_USERNAME }}
-      acr_password: ${{ secrets.ACR_PASSWORD }}
+  # 2. Base infrastructure creation
+  base:
+    name: Base 🧱
+    needs: setup
+    environment: ${{ needs.setup.outputs.environment }}
+    outputs:
+      environment: ${{ env.ENVIRONMENT }}
+    runs-on: [self-hosted, EXIP, infrastructure]
     steps:
+      - name: Pre-production 💫
+        if: contains('["development", "staging"]', env.TARGET)
+        run: echo "TYPE=Preproduction" >> $GITHUB_ENV
+
+      - name: Production 💫
+        if: ${{ 'prod' == env.TARGET }}
+        run: echo "TYPE=Production" >> $GITHUB_ENV
+
+      - name: Tags 🏷️
+        run: echo TAGS='Environment=${{ env.TYPE }}' \
+          'Product=${{ env.PRODUCT }}' \
+          'Team=development' >> $GITHUB_ENV
+
+      - name: Login 🔐
+        uses: azure/login@v1
+        with:
+          creds: ${{ secrets.AZURE_CREDENTIALS }}
+
+      - name: Azure defaults ✨
+        uses: Azure/cli@v1.0.7
+        with:
+          inlineScript: |
+            # Basic
+            az configure --defaults location=${{ vars.REGION }}
+            az configure --defaults group=rg-${{ env.PRODUCT }}-${{ env.TARGET }}-${{ vars.VERSION }}
+
+      - name: Resource group 🏗️
+        uses: Azure/cli@v1.0.7
+        with:
+          inlineScript: |
+            az group create \
+            --name rg-${{ env.PRODUCT }}-${{ env.TARGET }}-${{ vars.VERSION }} \
+            --tags ${{ env.TAGS }}
+
+      - name: Log analytics workspace 📝
+        uses: Azure/cli@v1.0.7
+        with:
+          inlineScript: |
+            az monitor log-analytics workspace create \
+            --name log-${{ env.PRODUCT }}-${{ env.TARGET }}-${{ vars.VERSION }} \
+            --capacity-reservation-level 100 \
+            --ingestion-access Disabled \
+            --query-access Disabled \
+            --quota 0.05 \
+            --retention-time ${{ vars.LOG_RETENTION_DAY }} \
+            --sku Free \
```

Comment:
Is it worth having the SKU in a var?

Reply:
Good shout, have introduced `LOG_ANALYTICS_PLAN`